app/api/global-message/route.tsx

```diff
@@ -0,0 +1,28 @@
+import {NextRequest, NextResponse} from "next/server"
+import {buildHeaders} from "@/lib/drupal/utils"
+import {GraphQLClient} from "graphql-request"
+import {getSdk} from "@/lib/gql/__generated__/queries"
+import type {RequestConfig} from "graphql-request/src/types"
+
+export const GET = async (request: NextRequest) => {
+  const graphqlClient = (requestConfig: RequestConfig = {}, isPreviewMode?: boolean) => {
+    requestConfig.headers = buildHeaders(requestConfig.headers as HeadersInit, isPreviewMode)
+    const client = new GraphQLClient(process.env.NEXT_PUBLIC_DRUPAL_BASE_URL + "/graphql", {
+      ...requestConfig,
+      next: {
+        revalidate: 5, // revalidate every 5 seconds to catch updates as soon as possible.
```

Comment:
I'm not sure that this is the right number here; there may be a better balance between BE requests and performance.  I just set it to 5 sec for testing purposes.

Reply:
This definitely is not good. This will cause every single page load to fetch every bit of data from Drupal. Resulting in millions of requests and slow page loads.

Reply:
Can I narrow this down so it only gets the config page for the global message instead of everything?

Reply:
That will still increase requests for nearly every single page every user visits because it will fetch new config page data every 5 seconds.